### PR TITLE
Add facility column and filter to results page

### PR DESCRIPTION
**Pull request by mehansen**
# FRONTEND PULL REQUEST

## Related Issue

The first step to completing #3230 All facilities view for results page. I wanted to break them up since adding the All facilities option will require some changes to the queries to handle either multiple filter facilities or no filter facilities

## Changes Proposed

Copied from notes on #3230 

> - add facility column to results page
> - add testing facility filter dropdown to results page, visible if user has access to > 1 facility
>     - default option is the facility selected in the header
>     - options include all the facilities the user has access to
>     - _[addressing in separate PR]_ admins have an additional option of an 'All facilities' filter
>     - no -Select- option
> - clicking 'Clear filters' button will still reset all filters to default (meaning facility will go back to what's selected in the header)
> - switching facility in the header also still resets all filters to default
> 

## Additional Information

- unsure of css best practices - would love feedback on if I set the padding for the table in the correct way
- was hoping to refactor DetachedTestResultsList props to not need filterFacility and activeFacility, but the csv download popup has different copy if we're filtering vs using the default settings

## Testing

- Pull branch or check it out in Dev and view results page. Verify filter behavior matches what's listed above
 
## Screenshots / Demos
Paired with Kenny on the column widths and padding changes.
Screen Shot 2022-04-13 at 3 20 15 PM

### Design
- [ ] Any UI/UX changes have a designer as a reviewer, and changes have been approved
- [ ] Any large-scale changes have been deployed to `test`, `dev`, or `pentest` and smoke-tested by both the engineering and design teams

### Content
- [ ] Any content changes have been approved by content team

### Support
- [ ] Any changes that might generate new support requests have been flagged to the support team
- [ ] Any changes to support infrastructure have been demo'd to support team

### Security
- [ ] Changes with security implications have been approved by a security engineer (changes to  authentication, encryption, handling of PII, etc.)
- [ ] Any dependencies introduced have been vetted and discussed

### CODE REVIEW

**Comment by mehansen** [comment]
Questions for @Rebecca-LM:
* If the user has access to only one facility, they don't see the "Testing facility" filter and the "Facility" column is the first time we reference facility on the page. Would it be better to change the column name to "Testing facility" for everyone?
* The "Download test results" popup has text that says "Download results [without any/with current] filters applied?" depending on whether the user has selected a filter. With my changes it will use the 'without any' text with the default facility filter and will use 'with current' once a user has actively selected a facility or other filter. Any thoughts on this behavior?
Screen Shot 2022-04-13 at 3 25 34 PM

**Comment by Rebecca-LM** [comment]
> Questions for @Rebecca-LM:
> 
> * If the user has access to only one facility, they don't see the "Testing facility" filter and the "Facility" column is the first time we reference facility on the page. Would it be better to change the column name to "Testing facility" for everyone?
> * The "Download test results" popup has text that says "Download results [without any/with current] filters applied?" depending on whether the user has selected a filter. With my changes it will use the 'without any' text with the default facility filter and will use 'with current' once a user has actively selected a facility or other filter. Any thoughts on this behavior?
>   Screen Shot 2022-04-13 at 3 25 34 PM

@mehansen all good questions. I think it's a good idea to call it "Testing facility" for the column name. We use "facility" a lot by itself, so our users are familiar with it and see it all the time, but the rule we should stick to is "testing facility" is the official/formal name for these that appears on headers when possible/on first mention..

My issue with the filter language is that we don't actually call them filters anywhere. I've brought this up before, but ideally we have a sort of section label that says "Filters" with all the options below it. It should probably not say "Search by name" but just "Name" since we don't give instructions on what to do with any of the rest. (If needed helper text could go in the name box itself.)

Probably not helpful for right now, but I did want to flag this again for you to keep in mind in the future! I think a good solution for now is to include the word search in the popup, so  "Download results [without any/with current] search filters applied?" Lmk what you think, I'm hoping it'll be more clear we're talking about search queries here!

File: frontend/src/app/testResults/TestResultsList.test.tsx [hunk 1]
```diff
@@ -1119,6 +1228,24 @@ describe("TestResultsList", () => {
...
+      expect(await screen.findByText("Colleer, Barde X")).toBeInTheDocument();
+      expect(
+        await screen.findByRole("option", { name: "Facility 1" })
+      ).toBeInTheDocument();
+      userEvent.selectOptions(screen.getByLabelText("Testing facility"), ["2"]);
+      expect(await screen.findByText("Clarkson, Lewis")).toBeInTheDocument();
+      expect(
+        screen.queryByText("Cragell, Barb Whitaker")
+      ).not.toBeInTheDocument();
+      expect(screen.queryByText("Colleer, Barde X")).not.toBeInTheDocument();
```
Comment (nickclyde) [inline frontend/src/app/testResults/TestResultsList.test.tsx:1247]: Thanks for adding this test!

File: frontend/src/app/testResults/TestResultsList.tsx [hunk 2]
```diff
@@ -508,6 +518,20 @@ export const DetachedTestResultsList = ({
                   defaultSelect
                   onChange={setFilterParams("role")}
                 />
+                {validFacilities && validFacilities.length > 1 ? (
```
Comment (nickclyde) [inline frontend/src/app/testResults/TestResultsList.tsx:521]: Awesome, I like that it only appears when the user has access to more than one facility!

**Review by nickclyde — APPROVED** [review]
Awesome work here! Tested in dev, works perfectly. :shipit:

File: frontend/src/app/testResults/TestResultsList.tsx (hunk 2 above)
Comment (mehansen) [reply frontend/src/app/testResults/TestResultsList.tsx:521]: Credit goes to @kenieh for that one :)

**Review by kenieh — APPROVED** [review]
LGTM

**Review by zdeveloper — APPROVED** [review]
LGTM! Thank you!

**Review by emmastephenson — APPROVED** [review]
LGTM!

**Comment by kenieh** [comment]
@mehansen @emmastephenson @jayna-SkylightDigital would it be feasible to have the "Testing facility" column only appear when the SR user has selected "all facilities" option in the "Testing facilities" filter? It's the only use case in which the facility column is useful. Otherwise, the facility column just repeats the same facility across all results and creates data clutter in the table.

**Comment by mehansen** [comment]
@kenieh Yes, this is possible but I'm not sure what the experience would feel like to have the columns resize when someone updates the filter. It's not a huge lift for me to try it out and show it to you though. We could also try hiding the column for everyone who is not an org admin, since only they can see the all facilities option.

**Pull request merged**